### PR TITLE
Add configuration of HTTPConnectionFactory with PEM certs

### PR DESCRIPTION
**Pull request by fbruton**

### CODE REVIEW

File: src/main/java/emissary/client/HTTPConnectionFactory.java [hunk 1]
```diff
@@ -183,16 +195,48 @@ private static KeyStore buildStore(final String path, final char[] pazz, final S
...
+        } else {
+            try (final InputStream is = Files.newInputStream(filePath)) {
+                keyStore.load(is, pazz);
+            }
         }
         return keyStore;
     }
 
+    protected static boolean isPemCert(Path filePath) throws IOException {
+        DataInputStream reader = new DataInputStream(Files.newInputStream(filePath));
```
Comment (dev-mlb) [inline src/main/java/emissary/client/HTTPConnectionFactory.java:215]: try-with-resources?

**Review by dev-mlb — CHANGES_REQUESTED** [review]

File: src/main/java/emissary/client/HTTPConnectionFactory.java (hunk 1 above)
Comment (fbruton) [reply src/main/java/emissary/client/HTTPConnectionFactory.java]: Good catch

**Review by dev-mlb — APPROVED** [review]

File: src/main/java/emissary/client/HTTPConnectionFactory.java [hunk 2]
```diff
@@ -183,16 +195,50 @@ private static KeyStore buildStore(final String path, final char[] pazz, final S
...
 
+    protected static boolean isPemCert(Path filePath) throws IOException {
+        byte[] beginBytes;
+        try (DataInputStream reader = new DataInputStream(Files.newInputStream(filePath))) {
+            int totalNumBytes = reader.available();
+            if (totalNumBytes <= BEGIN_CERT.length() + END_CERT.length()) {
+                return false;
+            }
+            beginBytes = new byte[BEGIN_CERT.length()];
+            reader.read(beginBytes);
```
Comment (cfkoehler) [inline src/main/java/emissary/client/HTTPConnectionFactory.java:222]: nit:
Not sure how much this matters in the real world so will defer to you more senior devs. Sonar complains about not checking the return value of the `read` call.
Would a `reader.readFully` make more sense here?
```suggestion
            reader.readFully(beginBytes);
```

**Review by cfkoehler — APPROVED** [review]

File: src/main/java/emissary/client/HTTPConnectionFactory.java [hunk 3]
```diff
@@ -183,16 +195,50 @@ private static KeyStore buildStore(final String path, final char[] pazz, final S
...
+        }
```
Comment (cfkoehler) [inline src/main/java/emissary/client/HTTPConnectionFactory.java:223]: I need to familiarize myself with `try with resources` more but should it still have a catch? Or just leave it up to the calling function to handle?

File: src/main/java/emissary/client/HTTPConnectionFactory.java (hunk 2 above)
Comment (jpdahlke) [reply src/main/java/emissary/client/HTTPConnectionFactory.java:222]: I think this makes sense, unlikely to matter in practice but simpler code.

File: src/main/java/emissary/client/HTTPConnectionFactory.java (hunk 3 above)
Comment (jpdahlke) [reply src/main/java/emissary/client/HTTPConnectionFactory.java:223]: Is this in reference to converting to readFully when it throws an IOException or in general? I believe that the calling method would just handle the exception in either case. Once the reader is out of scope of the try it closes (so before any catch/finally).

File: src/main/java/emissary/client/HTTPConnectionFactory.java (hunk 2 above)
Comment (fbruton) [reply src/main/java/emissary/client/HTTPConnectionFactory.java]: Updated

**Pull request merged**